From 352df35c124104586ee4c977556cb50ce720af62 Mon Sep 17 00:00:00 2001
From: Peter Jas <necmon@yahoo.com>
Date: Mon, 30 May 2016 12:54:02 +0300
Subject: [PATCH] Fix ucol_setMaxVariable detection for Gentoo Linux (#5309)

The issue was the symbol is exported by the ICU lib. Including headers
was not enough. The linker requires the libraries to succeed.

With this fix, CoreCLR successfully builds on Gentoo Linux 100%.
Tested with LXC gentoo container on Ubuntu machine.

Steps to configure and build:
https://gist.github.com/jasonwilliams200OK/1a2e2c0e904ffa95faf6333fcd88d9b8

Fix #5160
---
 src/corefx/System.Globalization.Native/configure.cmake | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/corefx/System.Globalization.Native/configure.cmake b/src/corefx/System.Globalization.Native/configure.cmake
index 4250c7c..138127e 100644
--- a/coreclr-1.0.0/src/corefx/System.Globalization.Native/configure.cmake
+++ b/coreclr-1.0.0/src/corefx/System.Globalization.Native/configure.cmake
@@ -8,11 +8,18 @@ CHECK_CXX_SOURCE_COMPILES("
     int main() { UDateFormatSymbolType e = UDAT_STANDALONE_SHORTER_WEEKDAYS; }
 " HAVE_UDAT_STANDALONE_SHORTER_WEEKDAYS)
 
+if(NOT CLR_CMAKE_PLATFORM_DARWIN)
+    set(CMAKE_REQUIRED_LIBRARIES ${ICUUC} ${ICUI18N})
+else()
+    set(CMAKE_REQUIRED_LIBRARIES ${ICUCORE})
+endif()
+
 check_symbol_exists(
     ucol_setMaxVariable
     "unicode/ucol.h"
     HAVE_SET_MAX_VARIABLE)
 
+unset(CMAKE_REQUIRED_LIBRARIES)
 unset(CMAKE_REQUIRED_INCLUDES)
 
 configure_file(
